ui/src/chat/ChatSearch/ChatSearch.tsx

```diff
@@ -0,0 +1,208 @@
+import cn from 'classnames';
+import { VirtuosoHandle } from 'react-virtuoso';
+import MagnifyingGlassIcon from '@/components/icons/MagnifyingGlassIcon';
+import X16Icon from '@/components/icons/X16Icon';
+import useDebounce from '@/logic/useDebounce';
+import useMedia, { useIsMobile } from '@/logic/useMedia';
+import { isTalk } from '@/logic/utils';
+import React, {
+  ChangeEvent,
+  KeyboardEvent,
+  PropsWithChildren,
+  useCallback,
+} from 'react';
+import { useNavigate, useParams } from 'react-router';
+import { Link } from 'react-router-dom';
+import * as Dialog from '@radix-ui/react-dialog';
+import { disableDefault } from '@/logic/utils';
+import { useChatSearch } from '@/state/chat';
+import bigInt from 'big-integer';
+import ChatSearchResults from './ChatSearchResults';
+
+interface RouteParams {
+  chShip: string;
+  chName: string;
+  query: string;
+  [key: string]: string;
+}
+
+type ChatSearchProps = PropsWithChildren<{
+  whom: string;
+  root: string;
+  placeholder: string;
+}>;
+
+export default function ChatSearch({
+  whom,
+  root,
+  placeholder,
+  children,
+}: ChatSearchProps) {
+  const navigate = useNavigate();
+  const { query } = useParams<RouteParams>();
```

Comment:
It might be a good idea to store the most recent query somewhere outside of the route in case the user wants to look through a few different results without having to search again.

Reply:
like a recent searches or something? I think that's a good idea, we def need a follow-up with something like that and actually using the mentions because the mocks had those and I'm sure people would appreciate